contracts/TwoUserMultisig.sol

```diff
@@ -1,74 +1,132 @@
+        _;
+    }
+
     constructor(address _owner1, address _owner2) {
         owner1 = _owner1;
         owner2 = _owner2;
     }
 
-	bytes4 constant EIP1271_SUCCESS_RETURN_VALUE = 0x1626ba7e;
-
-	modifier onlyBootloader() {
-		require(msg.sender == BOOTLOADER_FORMAL_ADDRESS, "Only bootloader can call this method");
-		// Continure execution if called from the bootloader.
-		_;
-	}
-
-	function validateTransaction(Transaction calldata _transaction) external payable override onlyBootloader {
-		_validateTransaction(_transaction);
-	}
+    function validateTransaction(
+        bytes32,
+        bytes32 _suggestedSignedHash,
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        _validateTransaction(_suggestedSignedHash, _transaction);
+    }
 
-	function _validateTransaction(Transaction calldata _transaction) internal {
+    function _validateTransaction(
+        bytes32 _suggestedSignedHash,
+        Transaction calldata _transaction
+    ) internal {
         // Incrementing the nonce of the account.
         // Note, that reserved[0] by convention is currently equal to the nonce passed in the transaction
-        NONCE_HOLDER_SYSTEM_CONTRACT.incrementNonceIfEquals(_transaction.reserved[0]);
-        bytes32 txHash = _transaction.encodeHash();
-
-        require(isValidSignature(txHash, _transaction.signature) == EIP1271_SUCCESS_RETURN_VALUE);
-	}
+        SystemContractsCaller.systemCall(
+            uint32(gasleft()),
+            address(NONCE_HOLDER_SYSTEM_CONTRACT),
+            0,
+            abi.encodeCall(
+                INonceHolder.incrementMinNonceIfEquals,
+                (_transaction.reserved[0])
+            )
+        );
+
+        bytes32 txHash;
+        // While the suggested signed hash is usually provided, it is generally
+        // not recommended to rely on it to be present, since in the future
+        // there may be tx types with no suggested signed hash.
+        if (_suggestedSignedHash == bytes32(0)) {
+            txHash = _transaction.encodeHash();
+        } else {
+            txHash = _suggestedSignedHash;
+        }
 
-	function executeTransaction(Transaction calldata _transaction) external payable override onlyBootloader {
-		_executeTransaction(_transaction);
-	}
+        require(
+            isValidSignature(txHash, _transaction.signature) ==
+                EIP1271_SUCCESS_RETURN_VALUE
+        );
+    }
 
-	function executeTransactionFromOutside(Transaction calldata _transaction) external payable {
-		_validateTransaction(_transaction);
-		_executeTransaction(_transaction);
-	}
+    function executeTransaction(
+        bytes32,
+        bytes32,
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        _executeTransaction(_transaction);
+    }
 
-	function _executeTransaction(Transaction calldata _transaction) internal {
+    function _executeTransaction(Transaction calldata _transaction) internal {
```

Comment:
In the original tutorial the following code is mentioned:

```ts
function _execute(Transaction calldata _transaction) internal {
    address to = address(uint160(_transaction.to));
    uint256 value = _transaction.reserved[1];
    bytes memory data = _transaction.data;

    if(to == address(DEPLOYER_SYSTEM_CONTRACT)) {
        // We allow calling ContractDeployer with any calldata
        SystemContractsCaller.systemCall(
            uint32(gasleft()),
            to,
            uint128(_transaction.reserved[1]), // By convention, reserved[1] is `value`
            _transaction.data
        );
    } else {
        bool success;
        assembly {
            success := call(gas(), to, value, add(data, 0x20), mload(data), 0, 0)
        }
        require(success);
    }
}
```

Could you please try using this one?

Reply:
Fixed in 55fbeee3b909bb5e0a907a9efc0deea0f67ec8f6